c++11/test/in_memory_recorder.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include <mutex>
+#include <stdexcept>
+#include <vector>
+#include "../src/recorder.h"
+
+namespace lightstep {
+// InMemoryRecorder is used for testing only.
+class InMemoryRecorder : public Recorder {
+ public:
+  void RecordSpan(collector::Span&& span) noexcept override {
+    std::lock_guard<std::mutex> lock_guard{mutex_};
+    spans_.emplace_back(std::move(span));
+  }
+
+  std::vector<collector::Span> spans() const {
+    std::lock_guard<std::mutex> lock_guard{mutex_};
+    return spans_;
+  }
+
+  size_t size() const {
```

Comment:
That class is only used for testing. Where it's invoked, the size method is only called from a single threaded context. I added the synchronization just to make the class general-purpose enough to be usable in multi-threaded tests if they're ever added.